homeassistant/components/ccm15/climate.py

```diff
@@ -0,0 +1,153 @@
+import logging
+from typing import Any
+
+from ccm15 import CCM15DeviceState
+
+from homeassistant.components.climate import (
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    PRECISION_WHOLE,
+    SWING_OFF,
+    SWING_ON,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import CONST_CMD_FAN_MAP, CONST_CMD_STATE_MAP, DOMAIN
+from .coordinator import CCM15Coordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up all climate."""
+    coordinator: CCM15Coordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    ac_data: CCM15DeviceState = coordinator.data
+    entities = [
+        CCM15Climate(coordinator.get_host(), ac_index, coordinator)
+        for ac_index in ac_data.devices
+    ]
+    async_add_entities(entities)
+
+
+class CCM15Climate(CoordinatorEntity[CCM15Coordinator], ClimateEntity):
+    """Climate device for CCM15 coordinator."""
+
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_has_entity_name = True
+    _attr_target_temperature_step = PRECISION_WHOLE
+    _attr_hvac_modes = [
+        HVACMode.OFF,
+        HVACMode.HEAT,
+        HVACMode.COOL,
+        HVACMode.DRY,
+        HVACMode.AUTO,
+    ]
+    _attr_fan_modes = [FAN_AUTO, FAN_LOW, FAN_MEDIUM, FAN_HIGH]
+    _attr_swing_modes = [SWING_OFF, SWING_ON]
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE
+        | ClimateEntityFeature.FAN_MODE
+        | ClimateEntityFeature.SWING_MODE
+    )
+    _attr_name = None
+
+    def __init__(
+        self, ac_host: str, ac_index: int, coordinator: CCM15Coordinator
+    ) -> None:
+        """Create a climate device managed from a coordinator."""
+        super().__init__(coordinator)
+        self._ac_host: str = ac_host
+        self._ac_index: int = ac_index
+        self._attr_unique_id = f"{self._ac_host}.{self._ac_index}"
+        self._attr_device_info = DeviceInfo(
+            identifiers={
+                # Serial numbers are unique identifiers within a specific domain
+                (DOMAIN, f"{self._ac_host}.{self._ac_index}"),
+            },
+            name=str(self._ac_index),
+            manufacturer="Midea",
+            model="CCM15",
+        )
+
+    @property
+    def current_temperature(self) -> int | None:
+        """Return current temperature."""
+        if data := self.coordinator.get_ac_data(self._ac_index):
+            return data.temperature
+        return None
```

Comment:
```suggestion
    @property
    def data(self) -> CCM15DeviceState | None:
        """Return device data."""
        return self.coordinator.get_ac_data(self._ac_index)

    @property
    def current_temperature(self) -> int | None:
        """Return current temperature."""
        if data := self.data:
            return data.temperature
        return None
```
Hmm, when I looked back at the code I saw this as an improvement, after writing this I am not sure. I think this approach adds a bit of abstraction on how to get the data (as you don't need to place the whole `coordinator.get_ac_data` for every property)

Feel free to ignore, but maybe its an idea

Reply:
I like it